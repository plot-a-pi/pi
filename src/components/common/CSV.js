import React from 'react';
import { PropTypes } from 'prop-types';
import { CSVLink } from 'react-csv';

const CSV = ({ csvData, header1, header2, children }) => {
  //csvData takes form array of couplets
  //header1 & header2 sit at the top of the file eg circumfrence diameter

  const csvDataPreparedForHeaders = csvData.map(datum => {
    return ({ 'x' : datum[0], 'y' : datum[1] });
  });

  const headers = [
    { label: header1, key: 'x' },
    { label: header2, key: 'y' }
  ];
  return (
    <CSVLink data={csvDataPreparedForHeaders} headers={headers} target='_blank'>
      {children}
    </CSVLink>
    
  );
};

CSV.propTypes = {
  csvData : PropTypes.array.isRequired,
  header1 : PropTypes.string,
  header2 : PropTypes.string,
  children : PropTypes.nodes
};

export default CSV;
