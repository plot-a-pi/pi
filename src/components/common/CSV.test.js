import React from 'react';
import { shallow } from 'enzyme';
import CSV from './CSV';

describe('CSV component', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(<CSV csvData={{ data: [[1, 1]] }} header1={'test'} header2={'snapshot'} />);
    expect(wrapper).toMatchSnapshot();
  });
});
