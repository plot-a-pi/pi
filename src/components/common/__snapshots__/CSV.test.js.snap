// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CSV component matches a snapshot 1`] = `
<CSVLink
  asyncOnClick={false}
  data={
    Array [
      Object {
        "x": 1,
        "y": 1,
      },
    ]
  }
  filename="generatedBy_react-csv.csv"
  headers={
    Array [
      Object {
        "key": "x",
        "label": "test",
      },
      Object {
        "key": "y",
        "label": "snapshot",
      },
    ]
  }
  separator=","
  target="_blank"
  uFEFF={true}
/>
`;
